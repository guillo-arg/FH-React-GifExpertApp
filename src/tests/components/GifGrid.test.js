import { GifGrid } from "../../components/GifGrid";
import {shallow} from 'enzyme';
import {useFetchGifs}  from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs")

const category = 'goku';

describe('pruebas sobre <GifGrid />', () => {

  test('Debe construirse correctamente', () => {
    useFetchGifs.mockReturnValue({
        data: [],
        loading: true
    });

    const wrapper = shallow(<GifGrid  category={category}/>);
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe mostrar items cuando se cargan imagenes', () => {
    const gifs = [{
        id: 'abc',
        url: 'http://localhost',
        title: 'title'
    }]

    useFetchGifs.mockReturnValue({
        data: gifs,
        loading: false
    });

    const wrapper = shallow(<GifGrid  category={category}/>);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);


  });
  
  
});
