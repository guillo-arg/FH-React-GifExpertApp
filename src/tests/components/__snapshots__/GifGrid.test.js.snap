// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas sobre <GifGrid /> Debe construirse correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    goku
  </h3>
  <p
    className="animate__animated animate__flash"
  >
     Loading 
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`pruebas sobre <GifGrid /> Debe mostrar items cuando se cargan imagenes 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    goku
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="abc"
      key="abc"
      title="title"
      url="http://localhost"
    />
  </div>
</Fragment>
`;
